### DISCUSSION TITLE
One more ws connection came from nowhere

### DISCUSSION THREAD

**Discussion by qiulang** [socketio/socket.io#4468] [Q&A]
I find I can trigger an unexpected socket.io connection under some circumstance

![2EIO](https://user-images.githubusercontent.com/516243/189087811-a3c6d459-a60b-4ad1-989d-b8d9dbca5f51.png)

This is my client code (4.5.2)

```
    const socket = io(wsServer,{
        transports: ['websocket'],
        auth:{token,uid,client}
    })
    socket.on("connect", () => {
        log(`debug 2nd connection ${socket.id}:${token}`)
        ...
    })
```

But from console I only see my log "debug 2nd connection"  **once**. I can't figure out where does the 2nd connection come from!
But I find under that circumstance if I set the function which will be called to a "real" function I can make the unexpected connection disappear:  e.g. I change the code from `let aCB` to `let aCB = ()=>{}`  to fix it.

The `aCB` is a callback function my user may provide. If they don't provide it and I don't set it to noop or check it explicitly before calling it, the unexpected ws connection will show up.

But why ? Any idea ?

**Comment by darrachequesne** [comment]
Hi! Could you please provide an example reproducing the issue? You can fork the example here: https://github.com/socketio/socket.io-fiddle

Without that, I'm afraid we won't be able to track down the issue. Thanks!

**Reply by qiulang** [reply]
Yes I will try to reproduce the issue based on https://github.com/socketio/socket.io-fiddle

**Reply by qiulang** [reply]
@darrachequesne when try to reproduce this problem, I hit another error, can you take a look https://github.com/darrachequesne/notepack/issues/29 ?

I have hit the problem with admin-ui before https://github.com/socketio/socket.io-admin-ui/issues/48

I have not been able to reproduce my problem based on a simple project template like https://github.com/socketio/socket.io-fiddle. I think that is because my problem involved with webpack bundled code. Without that I can't reproduce the problem, but webpack setting is just beyond me. 😅
